apps/script/HoC2023AiGenerateWeights.py

```diff
@@ -1,136 +1,159 @@
 # Script to generate the associated output weights contained in files like cached-spacy-background-map.json
-# that are used to calculate final effects output HoC2023
+# that are used to calculate final effects output HoC2023.
 # Run this Python script from the code-dot-org root directory.
+# python apps/script/HoC2023AiGenerateWeights.py
+# At the bottom of this file are background effects, palettes and foreground effects in the order they appear in the dropdowns.
+# When this file is run and the three cached data files are replaced in apps/static/dance/ai/model,
+# please update comments lists below with output of this script.
+
 import spacy
-import json
+from HoC2023AiHelperFunctions import *
 
 # Load the spaCy natural language processing model - English tokenizer, tagger, parser and NER (named entity recognition)
 nlp = spacy.load("en_core_web_lg")
 
-ai_inputs_file = open('apps/static/dance/ai/ai-inputs.json')
-emojis_data = json.load(ai_inputs_file)
-emojis_map = {}
-emojis_list = []
-for emoji in emojis_data['items']:
-    name = emoji['name']
-    id = emoji['id']
-    emojis_map[name] = id
-    emojis_list.append(name)
+# Get the list of emoji names and their corresponding emoji ids
+emoji_names_list, emojis_map = get_ai_emoji_inputs()
+print(emoji_names_list)
```

Comment:
Sorry if I'm having a little trouble following this, but are these print statements supposed to be here?

Reply:
Yes - print statements are copied and pasted as comments to ensure that the 'python_name' are matched correctly to the 'blockly_id' since they're defined in this file. And if the order of the dropdown options are changed or a dropdown option is added, then want to make sure 'python_name' is being mapped correctly.

Reply:
What does `emoji_names_list` look like? The comment that says "copy/paste here" on line 188, the first line below the comment is:

```
# BACKGROUND EFFECTS
# python_name | blockly_id | blockly_user_facing_name
```

which looks like its printed on line 63?

Reply:
My bad! Sorry - didn't look closely enough at where your comment was. Yes - I meant to delete these print statements. Thanks!